### PR TITLE
Fixed Firefox customization issue and added support for addons in my-resources

### PR DESCRIPTION
**Pull request by Skileau**
Firefox customization now works as announced in PR #107. 🦊 
Also, the user is now able to add his favourites addons to each new container with my-resources by two ways.

### With a list
It is possible to create a list in ~/.exegol/my-resources/setup/firefox/addons.txt with the links in Mozilla's addon shop :
image

> The last versions of the addons will then be downloaded and installed automatically.

### With xpi files
It is also possible to directly add .xpi files in the folder ~/.exegol/my-resources/setup/firefox/addons :
image

> The addons will then be installed from local .xpi files.

**Note** : both way can be used *simultaneously* by just creating the addons.txt and the addons folder.

### CODE REVIEW

File: sources/exegol/load_supported_setups.sh [hunk 1]
```diff
@@ -106,6 +106,18 @@ function run_user_setup() {
...
+function deploy_firefox_addons() {
+  if [ -d "$MY_Setup_PATH/firefox/" ]; then
+    if [ -d "$MY_Setup_PATH/firefox/addons" ]; then
+      ADDON_FOLDER="-D $MY_Setup_PATH/firefox/addons"
+    fi
+    if [ -f "$MY_Setup_PATH/firefox/addons.txt" ]; then
+      ADDON_LIST="-L $MY_Setup_PATH/firefox/addons.txt"
+    fi
+    python3 /opt/tools/firefox/user-setup.py $ADDON_LIST $ADDON_FOLDER
+  fi
```
Comment (Dramelac) [inline sources/exegol/load_supported_setups.sh:118]: If directories dont exist, why not created them ?

File: sources/install.sh [hunk 2]
```diff
@@ -3270,11 +3270,12 @@ function install_ctf-party() {
 
 function install_firefox() {
   colorecho "Installing firefox"
+  fapt firefox-esr
   mkdir /opt/tools/firefox
   mv /root/sources/firefox/* /opt/tools/firefox/
   python3 -m pip install -r /opt/tools/firefox/requirements.txt
   python3 /opt/tools/firefox/setup.py
-  add-test-command "firefox --version"
```
Comment (Dramelac) [inline sources/install.sh:3277]: We can keep both tests

**Review by Dramelac — CHANGES_REQUESTED** [review]

**Comment by ShutdownRepo** [comment]
test failure is unrelated to the PR and is due to runner's network instability
pipeline is ok, let's wait for @Dramelac's review

File: sources/exegol/load_supported_setups.sh [hunk 3]
```diff
@@ -124,7 +144,8 @@ deploy_tmux
 deploy_vim
 deploy_apt
 deploy_python3
+deploy_firefox_addons
```
Comment (Dramelac) [inline sources/exegol/load_supported_setups.sh:147]: Dont forget to submit a PR to the exegol-docs repository to add and explain your new feature.
@ShutdownRepo to review how we manage this requirement

**Review by Dramelac — APPROVED** [review]

File: sources/exegol/load_supported_setups.sh (hunk 3 above)
Comment (ShutdownRepo) [reply sources/exegol/load_supported_setups.sh:147]: Indeed, I didn't think about this. @Skileau happy to help if needed. Additions need to be made at [`source/exegol-image/my-resources.rst`](https://github.com/ThePorgs/Exegol-docs/blob/main/source/exegol-image/my-resources.rst).

Comment (Skileau) [reply sources/exegol/load_supported_setups.sh:147]: Yes, I had a quick look at how the doc is written this morning.
I will add a part for each of my recent PRs ;)

Comment (Skileau) [reply sources/exegol/load_supported_setups.sh:147]: ➡️ https://github.com/ThePorgs/Exegol-docs/pull/6
Do not hesitate to request changes, I tried to keep the other section's style.

Comment (ShutdownRepo) [reply sources/exegol/load_supported_setups.sh:147]: Picked up at https://github.com/ThePorgs/Exegol-docs/pull/7

**Comment by Dramelac** [comment]
@ShutdownRepo good for merge ?

**Comment by Skileau** [comment]
> @Skileau can you confirm custom cert trusting will be handled in another PR?

Yes, you can merge, I will create a new PR with cert trusting support in the future.
We will define together the structure of the code for this new PR.

**Pull request merged**